app/Models/Loggable.php

```diff
@@ -100,27 +133,65 @@ private function determineLogItemType($log)
      */
     public function logCheckin($target, $note)
     {
+        $settings = Setting::getSettings();
         $log = new Actionlog;
         $log->target_type = get_class($target);
-        $log->target_id = $target->id;
+
+        if ($target) {
+            $log->target_id = $target->id;
+        } else {
+            $target = null;
+        }
+
+
         if (static::class == LicenseSeat::class) {
             $log->item_type = License::class;
             $log->item_id = $this->license_id;
         } else {
             $log->item_type = static::class;
             $log->item_id = $this->id;
         }
+
+
         $log->location_id = null;
         $log->note = $note;
         $log->user_id = Auth::user()->id;
         $log->logaction('checkin from');
 
         $params = [
+            'target' => $target,
             'item' => $log->item,
             'admin' => $log->user,
-            'note' => $note
+            'note' => $note,
         ];
-        Setting::getSettings()->notify(new CheckinNotification($params));
+
+        // Get the admin email address we should be sending notifications to, if any
+        $recipient = new \App\Models\Recipients\AdminRecipient();
+
+        if (static::class == Asset::class) {
+            if ($settings->admin_cc_email!='') {
+                $recipient->notify(new CheckinAssetNotification($params));
+            }
+
+            $target->notify(new CheckinAssetNotification($params));
+
+        } elseif (static::class == Accessory::class) {
+            if ($settings->admin_cc_email!='') {
+                $recipient->notify(new CheckinAccessoryNotification($params));
+            }
+
+            $target->notify(new CheckinAccessoryNotification($params));
+
+        } elseif (static::class == LicenseSeat::class) {
+            if ($settings->admin_cc_email!='') {
+                $recipient->notify(new CheckinLicenseNotification($params));
+            }
+
+            $target->notify(new CheckinLicenseNotification($params));
+
+        }
+
+
```

Comment:
Following up on previous comment, wonder about extracting a determineTarget() method for both places, would avoid having to keep both in sync...